tests/Core/Dependency/Emitter/ClassDependencyEmitterTest.php

```diff
@@ -23,23 +23,25 @@ public function testApplyDependencies(): void
             __DIR__.'/Fixtures/Foo.php'
         );
 
-        self::assertCount(18, $deps);
-        self::assertContains('Foo\Bar:6 on Foo\BarExtends', $deps);
-        self::assertContains('Foo\Bar:6 on Foo\BarInterface1', $deps);
-        self::assertContains('Foo\Bar:6 on BarInterface2', $deps);
-        self::assertContains('Foo\Bar:8 on Foo\SomeTrait', $deps);
-        self::assertContains('Foo\Bar:10 on Foo\SomeParam', $deps);
-        self::assertContains('Foo\Bar:10 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:12 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:13 on SomeOtherClass', $deps);
-        self::assertContains('Foo\Bar:15 on Foo\SomeOtherParam', $deps);
-        self::assertContains('Foo\Bar:19 on Foo\SomeInstanceOf', $deps);
-        self::assertContains('Foo\Bar:21 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:23 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:26 on Some\NamespacedClass', $deps);
-        self::assertContains('Foo\Bar:30 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:32 on Foo\SomeClass', $deps);
-        self::assertContains('Foo\Bar:36 on Foo\string2', $deps);
-        self::assertContains('Foo\Bar:42 on Foo\SomeClass', $deps);
+        self::assertEqualsCanonicalizing([
+            'Foo\Bar:6 on Foo\BarExtends',
+            'Foo\Bar:6 on Foo\BarInterface1',
+            'Foo\Bar:6 on BarInterface2',
+            'Foo\Bar:8 on Foo\SomeTrait',
+            'Foo\Bar:10 on Foo\SomeParam',
+            'Foo\Bar:10 on Foo\SomeClass',
+            'Foo\Bar:12 on Foo\SomeClass',
+            'Foo\Bar:13 on SomeOtherClass',
+            'Foo\Bar:15 on Foo\SomeOtherParam',
+            'Foo\Bar:19 on Foo\SomeInstanceOf',
+            'Foo\Bar:21 on Foo\SomeClass',
+            'Foo\Bar:23 on Foo\SomeClass',
+            'Foo\Bar:26 on Some\NamespacedClass',
+            'Foo\Bar:30 on Foo\SomeClass',
+            'Foo\Bar:32 on Foo\SomeClass',
+            'Foo\Bar:36 on Foo\string2',
+            'Foo\Bar:38 on string', // todo: This is a bug in the parser for FQ "\string"
```

Comment:
Here we have a bug where `\string` is treated as class `string` instead of the scalar type `string.`